pkg/noun/events.c

```diff
@@ -314,6 +348,35 @@ _ce_image_stat(u3e_image* img_u, c3_w* pgs_w)
   }
 }
 
+/* _ce_ephemeral_open(): open or create ephemeral file
+*/
+static c3_o
+_ce_ephemeral_open(c3_i* eph_i)
+{
+  c3_i mod_i = O_RDWR | O_CREAT;
+  c3_c ful_c[8193];
+
+  snprintf(ful_c, 8192, "%s", u3P.dir_c);
+  c3_mkdir(ful_c, 0700);
+
+  snprintf(ful_c, 8192, "%s/.urb", u3P.dir_c);
+  c3_mkdir(ful_c, 0700);
+
+  snprintf(ful_c, 8192, "%s/.urb/ephemeral.bin", u3P.dir_c);
+  if ( -1 == (*eph_i = c3_open(ful_c, mod_i, 0666)) ) {
+    fprintf(stderr, "loom: ephemeral c3_open %s: %s\r\n", ful_c,
+            strerror(errno));
+    return c3n;
+  }
+
+  if ( ftruncate(*eph_i, u3a_bytes) < 0 ) {
```

Comment:
`u3a_bytes` is the compile-time maximum (8 GB). you can use `_ce_len(u3P.pag_w)` to for "`--loom` in bytes"

Reply:
that's better